rosbridge_server/scripts/rosbridge_websocket.py

```diff
@@ -55,184 +60,240 @@
+                                              if max_message_size is None
+                                              else max_message_size
+
+        unregister_timeout = self.get_parameter('unregister_timeout').value
+        RosbridgeWebSocket.unregister_timeout = RosbridgeWebSocket.unregister_timeout
+                                                if unregister_timeout is None
+                                                else unregister_timeout
+
+        bson_only_mode = self.get_parameter('retry_startup_delay').value
+        bson_only_mode = False if bson_only_mode is None
+                               else bson_only_mode
+
+        if RosbridgeWebSocket.max_message_size == "None":
+            RosbridgeWebSocket.max_message_size = None
+
+        # SSL options
+        certfile = self.get_parameter('certfile').value
+        keyfile = self.get_parameter('keyfile').value
+        # if authentication should be used
+        authenticate = self.get_parameter('authenticate').value
+        RosbridgeWebSocket.authenticate = False if authenticate is None else authenticate
+
+        port = self.get_parameter('port').value
+        port = 9090 if port is None else port
+        
+        address = self.get_parameter('address').value
+        address = "" if address is None else address
+
+        # Publisher for number of connected clients
+        # QoS profile with transient local durability (latched topic in ROS 1).
+        client_count_qos_profile = qos_profile_default
+        client_count_qos_profile.durability = QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL
+
+        RosbridgeWebSocket.client_count_pub = self.create_publisher(Int32, 'client_count',
+                                                                    qos_profile=client_count_qos_profile)
+        RosbridgeWebSocket.client_count_pub.publish(Int32(data=0))
+
+        # Get the glob strings and parse them as arrays.
+        topics_glob = self.get_parameter('topics_glob').value
+        topics_glob = '' if topics_glob is None else topics_glob
+
+        services_glob = self.get_parameter('topics_glob').value
+        services_glob = '' if services_glob is None else services_glob
+
+        params_glob = self.get_parameter('params_glob').value
+        params_glob = '' if params_glob is None else params_glob
+
+        RosbridgeWebSocket.topics_glob = [
+            element.strip().strip("'")
+            for element in topics_glob[1:-1].split(',')
+            if len(element.strip().strip("'")) > 0]
+        RosbridgeWebSocket.services_glob = [
+            element.strip().strip("'")
+            for element in services_glob[1:-1].split(',')
+            if len(element.strip().strip("'")) > 0]
+        RosbridgeWebSocket.params_glob = [
+            element.strip().strip("'")
+            for element in params_glob[1:-1].split(',')
+            if len(element.strip().strip("'")) > 0]
+
+        if "--port" in sys.argv:
```

Comment:
I would consider start using `argparse`.

Reply:
Maybe, that can be done in the original code too.

Reply:
Yes, this seems to be like a manually implemented `argparse`. I'd say it's out of scope of this PR.